pkg/sql/logictest/testdata/logic_test/hash_sharded_index

```diff
@@ -483,3 +483,36 @@ INSERT INTO sharded_index_with_nulls VALUES (1, NULL);
 
 statement ok
 DROP TABLE sharded_index_with_nulls;
+
+# Test that renaming a column which is a member of a hash sharded index works.
+subtest rename_column
+
+statement ok
+CREATE TABLE rename_column (
+    c0 INT,
+    c1 INT,
+    c2 INT,
+    PRIMARY KEY (c0, c1) USING HASH WITH BUCKET_COUNT = 8,
+    INDEX (c2) USING HASH WITH BUCKET_COUNT = 8
+);
+
+statement ok
+ALTER TABLE rename_column RENAME c2 TO c3;
```

Comment:
Do you need to insert/select from this table to ensure that everything works, or is just doing the rename enough?